Compiler error while building shared library in MSYS2/MINGW

* Build platform: Windows 10 MSYS2/MINGW
* Compiler: GCC 10.2.0 x86_64-w64-mingw32
* CMake version 3.18.4
* Ninja version 1.10.1
* Command used:

```sh
git clone --depth=1 https://github.com/KhronosGroup/glslang.git
cd glslang; mkdir bin; cd bin
MSYS2_ARG_CONV_EXCL="-DCMAKE_INSTALL_PREFIX=" \
${MINGW_PREFIX}/bin/cmake.exe \
    -G "Ninja" \
    -DCMAKE_INSTALL_PREFIX=${MINGW_PREFIX} \
    -DCMAKE_INSTALL_LIBDIR=lib \
    -DCMAKE_BUILD_TYPE=Release \
    -DBUILD_SHARED_LIBS=ON \
    ../
${MINGW_PREFIX}/bin/cmake.exe --build ./
```

With `-DBUILD_SHARED_LIBS=OFF` option, it is compiled fine.

* Compiler Output:

```c
[12/68] Building CXX object StandAlone/CMakeFiles/glslangValidator.dir/StandAlone.cpp.obj
FAILED: StandAlone/CMakeFiles/glslangValidator.dir/StandAlone.cpp.obj
F:\msys64\mingw64\bin\c++.exe -DENABLE_HLSL -DENABLE_OPT=0 -DGLSLANG_IS_SHARED_LIBRARY=1 -DGLSLANG_OSINCLUDE_WIN32 -I../StandAlone/../External -I../glslang/.. -Iinclude -I../SPIRV/.. -I../StandAlone -I../ -O3 -DNDEBUG -Wall -Wmaybe-uninitialized -Wuninitialized -Wunused -Wunused-local-typedefs -Wunused-parameter -Wunused-value -Wunused-variable -Wunused-but-set-parameter -Wunused-but-set-variable -fno-exceptions -Wno-reorder -fno-rtti -Werror=deprecated-copy -std=c++11 -MD -MT StandAlone/CMakeFiles/glslangValidator.dir/StandAlone.cpp.obj -MF StandAlone\CMakeFiles\glslangValidator.dir\StandAlone.cpp.obj.d -o StandAlone/CMakeFiles/glslangValidator.dir/StandAlone.cpp.obj -c ../StandAlone/StandAlone.cpp
In file included from ../StandAlone/DirStackFileIncluder.h:44,
                 from ../StandAlone/StandAlone.cpp:45:
F:/glslang/glslang/Public/ShaderLang.h:704:33: error: function 'const glslang::TType* glslang::TObjectReflection::getType() const' definition is marked dllimport
  704 |     GLSLANG_EXPORT const TType* getType() const { return type; }
      |                                 ^~~~~~~
[14/68] Building CXX object glslang/CMakeFiles/MachineIndependent.dir/MachineIndependent/glslang_tab.cpp.obj
ninja: build stopped: subcommand failed.
```

* Temporary fix:

```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index ada2b8f..c6460ea 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -293,7 +293,7 @@ function(glslang_only_export_explicit_symbols target)
     if(BUILD_SHARED_LIBS)
         target_compile_definitions(${target} PUBLIC "GLSLANG_IS_SHARED_LIBRARY=1")
         if(WIN32)
-            target_compile_definitions(${target} PRIVATE "GLSLANG_EXPORTING=1")
+            target_compile_definitions(${target} PUBLIC "GLSLANG_EXPORTING=1")
         else()
             target_compile_options(${target} PRIVATE "-fvisibility=hidden")
         endif()
```

I would be happy to provide any further information, if required.

--- a/glslang/Public/ShaderLang.h
+++ b/glslang/Public/ShaderLang.h
@@ -722,7 +722,7 @@ class TObjectReflection {
 public:
     GLSLANG_EXPORT TObjectReflection(const std::string& pName, const TType& pType, int pOffset, int pGLDefineType, int pSize, int pIndex);
 
-    GLSLANG_EXPORT const TType* getType() const { return type; }
+    const TType* getType() const { return type; }
     GLSLANG_EXPORT int getBinding() const;
     GLSLANG_EXPORT void dump() const;
     static TObjectReflection badReflection() { return TObjectReflection(); }

